### PR TITLE
improved readability, typo corrections, refreshed links, improved handling of First and Last, improved error handling, overhauled SortPassages

### PR DESCRIPTION
**Pull request by ace280**
Removed most one-letter variables, more expressive descriptions, small corrections of descriptions, some typo corrections
refreshed links as CITE Architecture moved to http://cite-architecture.org/

### CODE REVIEW

**Comment by ace280** [comment]
Please also have a look at https://github.com/ace280/gocite/tree/requestForChange

File: gocite.go [hunk 1]
```diff
@@ -661,17 +673,18 @@ func ExtractTextByID(id string, w Work) ([]TextAndID, error) {
...
+		if straw == needle {
 			return true
 		}
 	}
 	return false
 }
 
-func findStartEnd(s string) (start, end string, err error) {
-	urn := SplitCTS(s)
+func findStartEnd(URNString string) (start, end string, err error) {
```
Comment (ThomasK81) [inline gocite.go]: haystack not "hey, stack!"

File: gocite.go [hunk 2]
```diff
@@ -661,17 +673,18 @@ func ExtractTextByID(id string, w Work) ([]TextAndID, error) {
 	return selection, nil
 }
 
-func contains(s []string, e string) bool {
-	for _, a := range s {
-		if a == e {
+//contains returns true if the 'needle' string is found in the 'heystack' string slice
+func contains(heystack []string, needle string) bool {
```
Comment (ThomasK81) [inline gocite.go]: haystack not "hey, stack!"

File: gocite.go [hunk 3]
```diff
@@ -502,7 +514,7 @@ func before(value string, a string) (string, error) {
 }
 
 // ExtractTextByID extracts the textual information from a Passage or multiple Passages in a Work
-func ExtractTextByID(id string, w Work) ([]TextAndID, error) {
+func ExtractTextByID(id string, work Work) ([]TextAndID, error) {
```
Comment (ThomasK81) [inline gocite.go]: ctsid

File: gocite.go [hunk 4]
```diff
@@ -148,220 +150,230 @@ func IsCITEURN(s string) bool {
...
-// FindLastIndex returns the first Index of the Passages in a Work.
-// It also returns a bool whether it has found one.
-func FindLastIndex(w Work) (int, bool) {
-	for i := range w.Passages {
-		if w.Passages[i].Last.Exists {
-			return w.Passages[i].Last.Index, true
+// GetLastIndex returns the Passage.Last.Index of the first Passage in a Work
+//that has Passage.Last.Index.Exists set to true,
+//along with a bool whether it has found that index.
+func GetLastIndex(work Work) (int, bool) {
```
Comment (ThomasK81) [inline gocite.go]: Make the rename clear in the comments. Still keep "FindLastIndex" for legacy.

File: gocite.go [hunk 5]
```diff
@@ -148,220 +150,230 @@ func IsCITEURN(s string) bool {
...
-// FindFirstIndex returns the first Index of the Passages in a Work.
-// It also returns a bool whether it has found one.
-func FindFirstIndex(w Work) (int, bool) {
-	for i := range w.Passages {
-		if w.Passages[i].First.Exists {
-			return w.Passages[i].First.Index, true
+// GetFirstIndex returns the Passage.First.Index of the first Passage in a Work
+//that has Passage.First.Index.Exists set to true,
+//along with a bool whether it has found that index.
+func GetFirstIndex(work Work) (int, bool) {
```
Comment (ThomasK81) [inline gocite.go]: Make the rename clear in the comments. Still keep "FindFirstIndex" for legacy.

File: gocite.go [hunk 6]
```diff
@@ -148,220 +150,230 @@ func IsCITEURN(s string) bool {
...
-func GetNext(id string, w Work) Passage {
-	for i := range w.Passages {
-		if w.Passages[i].PassageID == id {
-			return w.Passages[w.Passages[i].Prev.Index]
+// GetNext returns the Passage after the Passage given the PassageID in given Work
+//(not the next in the work.Passages slice)
+func GetNext(passageID string, work Work) Passage {
+	for i := range work.Passages {
+		if work.Passages[i].PassageID == passageID {
+			return work.Passages[work.Passages[i].Next.Index]
```
Comment (ThomasK81) [inline gocite.go]: good spotting.

**Comment by ace280** [comment]
Implemented the changes as discussed. If you can please have a look so we can discuss tomorrow.

**Pull request merged**